dashboard/app/views/layouts/_small_footer.html.haml

```diff
@@ -1,24 +1,40 @@
-.small-footer
-  - if view_options[:has_i18n]
-    = form_tag(locale_url, method: :post, id: 'localeForm', style: 'margin-bottom: 0px;') do
-      = hidden_field_tag :return_to, request.url
-      = select_tag :locale, options_for_select(options_for_locale_select, locale), onchange: 'this.form.submit();'
-  %small
-    = link_to I18n.t('footer.copyright'), '#', class: 'copyright-link'
-    &nbsp;&nbsp;|&nbsp;&nbsp;
-    %a{:href => "#", :class => "more-link"}
-      = I18n.t('footer.more')
-      %i{:class => "fa fa-caret-up"}
+= javascript_include_tag 'react'
 
-#copyright-flyout
-  %p!= I18n.t('footer.thank_you')
-  %p= I18n.t('footer.help_from_html')
-  %p!= I18n.t('footer.art_from_html', current_year: Time.now.year)
-  %p= I18n.t('footer.powered_by_aws')
-  %p!= I18n.t('footer.trademark', current_year: Time.now.year)
+#page-small-footer
 
-%ul#more-menu
-  %li= link_to I18n.t('footer.support'), 'https://support.code.org'
-  %li= link_to I18n.t('footer.translate'), 'https://code.org/translate'
-  %li= link_to I18n.t('footer.tos'), 'https://code.org/tos'
-  %li= link_to I18n.t('footer.privacy'), 'https://code.org/privacy'
+:javascript
+
+  var reactProps = {
+    i18nDropdown: "#{URI.encode(i18n_dropdown.to_str) if view_options[:has_i18n]}",
+    copyrightInBase: true,
+    baseCopyrightString: "#{I18n.t('footer.copyright')}",
+    baseMoreMenuString: "#{I18n.t('footer.more')}",
+    menuItems: [
+      {
+        text: "#{I18n.t('footer.support')}",
+        link: "https://support.code.org"
+      },
+      {
+        text: "#{I18n.t('footer.translate')}",
+        link: "https://code.org/translate"
+      },
+      {
+        text: "#{I18n.t('footer.tos')}",
+        link: "https://code.org/tos"
+      },
+      {
+        text: "#{I18n.t('footer.privacy')}",
+        link: "https://code.org/privacy"
+      }
+    ],
+    copyrightStrings: {
+      thank_you: "#{URI.escape(I18n.t('footer.thank_you'))}",
+      help_from_html: "#{I18n.t('footer.help_from_html')}",
+      art_from_html: "#{URI.escape(I18n.t('footer.art_from_html', current_year: Time.now.year))}",
+      powered_by_aws: "#{I18n.t('footer.powered_by_aws')}",
+      trademark: "#{URI.escape(I18n.t('footer.trademark', current_year: Time.now.year))}"
+    }
+  };
+
+  window.dashboard.footer.render(React, reactProps,
+    document.getElementById('page-small-footer'));
```

Comment:
This latest changes moves a little bit more of the JS out (i.e. no React.createElement/React.render). Is there a clean way of having large blocks of ruby inside haml? If so, we could make this even less javascripty by creating our props hash in ruby and passing that to footer.render, i.e. something like
```
#page-small-footer
:ruby
 props = {
   ...
}

:javascript
  window.dashboard.footer.render(React, #{props}, document.getElementById('page-small-footer'))